### ISSUE TITLE
WCAG audit

### ISSUE THREAD

**Issue by EmreBahar0912** [EmreBahar0912/all-human-accessible-website#12]
WCAG audit uitvoeren op de contactpagina, gevonden problemen gedocumenteerd en verholpen.

**Comment by EmreBahar0912** [comment]
# Lighthouse test

## Score
<img width="531" height="213" alt="Image" src="https://github.com/user-attachments/assets/b98d3444-8486-4e3a-ad57-957720b60631" />

## Failed audits
<img width="770" height="442" alt="Image" src="https://github.com/user-attachments/assets/f0ddec2a-6709-438d-91ae-48ef6bc1567a" />

### Names and labels
- Afbeeldingen hebben geen alt attributen:
<img width="713" height="453" alt="Image" src="https://github.com/user-attachments/assets/8b3842c7-acd1-42c3-ac64-ff1496fb10d3" />

### Oplossing
- Alt attributen toegevoegd aan afbeeldingen:
<img width="863" height="556" alt="Image" src="https://github.com/user-attachments/assets/a85bd0d7-ed29-43a5-9005-1da8dde24637" />
<img width="705" height="328" alt="Image" src="https://github.com/user-attachments/assets/e12bf6dd-9c4d-47de-b430-f975b71f7028" />

### Contrast
- Geen goede kleurcontrast bij "Lees meer" linkjes:
<img width="739" height="307" alt="Image" src="https://github.com/user-attachments/assets/171bc860-9256-49e2-b573-4d3da433fd6d" />

### Oplossing
- Zwarte tekstkleur geven i.p.v. wit voor beter contrast:
<img width="89" height="30" alt="Image" src="https://github.com/user-attachments/assets/5673fab3-4367-41d3-a3c4-f629dbc92d32" />
<img width="433" height="616" alt="Image" src="https://github.com/user-attachments/assets/ccabf14b-5905-4bb2-80cc-27a3831e15cf" />

### Navigation
- Heading is niet op chronologische volgorde bij de footer:
<img width="745" height="358" alt="Image" src="https://github.com/user-attachments/assets/c32cdedc-97a1-4fec-a9d3-b69fcaad544c" />

### Oplossing
- Bij de footer heb ik h2 en h4 gebruikt, wat niet de juiste volgorde is voor headings, daarom heb ik de h4's veranderd naar h3's:
<img width="866" height="309" alt="Image" src="https://github.com/user-attachments/assets/411ca459-67a0-4c93-a227-d6bb1753d9d0" />
<img width="862" height="309" alt="Image" src="https://github.com/user-attachments/assets/27701a9b-611c-4659-8fa8-6731d7fe4f6f" />

## Passed audits
<img width="735" height="442" alt="Image" src="https://github.com/user-attachments/assets/ea2d2456-3842-458d-85c0-41c9cf6c7e3b" />
<img width="727" height="191" alt="Image" src="https://github.com/user-attachments/assets/fb2f00c6-4961-493b-b692-e506f66570ff" />

## Uitleg
1. **`[aria-hidden="true"] is not present on the document `<body>** – Er is geen verborgen content via aria-hidden op de body-tag, wat goed is voor toegankelijkheid.
2. **Buttons have an accessible name** – Alle knoppen hebben een naam die schermlezers kunnen lezen, zodat gebruikers ze makkelijk en snel kunnen herkennen.
3. **`[user-scalable="no"] is not used in the `<meta name="viewport">` element and the [maximum-scale] attribute is not less than 5`** – De website laat gebruikers in- en uitzoomen, wat fijn is voor mensen die grotere tekst willen.
4. **Document has a `<title>` element** – De pagina heeft een titel, die je in het tabblad ziet en belangrijk is voor vindbaarheid en toegankelijkheid.
5. **`<html>` element has a `[lang]` attribute** – De taal van de pagina is aangegeven, zodat schermlezers weten hoe ze tekst moeten uitspreken.
6. **`<html>` element has a valid value for its `[lang]` attribute** – De taal afkorting is correct ingevuld, bijvoorbeeld ‘en’ voor Engels.
7. **Links are distinguishable without relying on color** – Links zijn duidelijk zichtbaar, ook met een visuele beperking.
8. **Links have a discernible name** – Links hebben een duidelijke naam of tekst, zodat je weet waar je heen gaat als je erop klikt.
9. **Lists contain only `<li>` elements and script supporting elements (`<script>` and `<template>`).** – Lijsten mogen alleen lijstjes en script elementen bevatten.
10. **List items (`<li>`) are contained within `<ul>`, `<ol>` or `<menu>` parent elements.** – `<li>` moet altijd in een lijst zitten, dus niet los op de pagina.
11. **Touch targets have sufficient size and spacing.** – Knoppen en links moeten groot genoeg zijn en niet te dicht bij elkaar, zodat je ze makkelijk aan kunt klikken.
12. **Image elements do not have `[alt]` attributes that are redundant text.** – `alt` tekst moet nuttig zijn, niet een herhaling van wat je al ziet.

# Handmatige tests

## Keyboard
- Bij de header was de tab-volgorde niet logisch, hamburger-menu stond namelijk boven de logo zelf in HTML bestand volgorde, terwijl logo op de website bovenaan staat (op laptop versie dan). Het lijkt me juist handiger om van boven te beginnen, dus daarom heb ik de logo in het HTML bestand boven het hamburger-menu gezet, zodat de tab-volgorder logischer is want op laptop hoort het hamburger-menu te verdwijnen. Op laptop begin je nu dus boven, en op telefoon en tablet is de volgorde van links naar rechts.
Van regel 31 naar regel 17 verplaatst:
<img width="825" height="383" alt="Image" src="https://github.com/user-attachments/assets/33cac727-749f-4e57-ab79-05bfaf651042" />
<img width="821" height="383" alt="Image" src="https://github.com/user-attachments/assets/59234cb0-9cb2-4d94-a287-2b5194a43e88" />

- Het tabben bij hamburger menu ging ook niet soepel op telefoon en tablet formaat. Wanneer hamburger menu niet geopend was, dan kon je er nog steeds op tabben maar je zag helemaal niks gebeuren op je scherm als gebruiker zijnde, alleen links die veranderen links onderin het scherm. Dit heb ik opgelost met behulp van tab-index = -1 te gebruiken voor de linkjes in het hamburger-menu en aria-expanded = false te zetten op het hamburger-menu. En met behulp van JavaScript heb ik dit werkend kunnen maken:
<img width="826" height="384" alt="Image" src="https://github.com/user-attachments/assets/5d42a844-0afd-4a2e-9006-1b47a4654d28" />
<img width="603" height="250" alt="Image" src="https://github.com/user-attachments/assets/b440279b-e0b7-454b-8ccd-cc02ae6f575b" />

- Voor de rest gaat tabben in een goede volgorde op de website, met de carousel buttons wordt tabben overgeslagen als het op disabled staat en je dus niet verder kunt scrollen met de buttons bij het carousel. Dat vind ik een mooi om te zien.

## Screenreader
- Bij de screenreader test kwam ik een aantal problemen tegen, namelijk dat de meeste afbeeldingen geen alt attribute hebben en icons hebben geen aria-label gekregen. Dus dat heb ik aangepast:
<img width="861" height="99" alt="Image" src="https://github.com/user-attachments/assets/e4132c94-05c4-4811-8190-fb3773a64771" />
<img width="863" height="556" alt="Image" src="https://github.com/user-attachments/assets/a85bd0d7-ed29-43a5-9005-1da8dde24637" />
<img width="705" height="328" alt="Image" src="https://github.com/user-attachments/assets/e12bf6dd-9c4d-47de-b430-f975b71f7028" />

## Interactive elements
- Al mijn interactive elements hebben een cursor: pointer; zodat het duidelijk is dat wanneer je eroverheen hovert het een klikbaar element is. "Lees meer" links bij carousel en de linkjes bij het menu in de header hebben krijgen ook een donker groene kleur wanneer je eroverheen hovert met je muis, zodat het nog duidelijker is voor gebruikers dat het om interactive elements gaat.
<img width="77" height="35" alt="Image" src="https://github.com/user-attachments/assets/5656c83a-b654-44f9-b347-754e38b34b41" />
<img width="74" height="32" alt="Image" src="https://github.com/user-attachments/assets/393f225b-88ea-457e-b482-6166798a77ef" />
<img width="494" height="92" alt="Image" src="https://github.com/user-attachments/assets/856bf6ba-bd02-4869-af31-5a83f479d20e" />
<img width="468" height="96" alt="Image" src="https://github.com/user-attachments/assets/c34452b3-dd1e-47fe-a68f-38d125d15d33" />

- Alleen bij de footer is het niet duidelijk dat het om linkjes gaat bij de namen van de bestuursleden. Er is alleen een cursor pointer aanwezig, maar het zou beter zijn om ook een hover toe te voegen namelijk een underline:
<img width="568" height="295" alt="Image" src="https://github.com/user-attachments/assets/bbd349d6-c7a4-4d5f-890b-a4afd3534cf4" />
<img width="536" height="287" alt="Image" src="https://github.com/user-attachments/assets/59502c1b-c7c4-4054-99a7-86c676d01c9d" />
<img width="329" height="360" alt="Image" src="https://github.com/user-attachments/assets/371208fa-d1ac-4f9e-a868-b2d704e75865" />

## Headings and landmarks
- Bij de footer heb ik h2 en h4 gebruikt, wat niet de juiste volgorde is voor headings, daarom heb ik de h4's veranderd naar h3's:
<img width="866" height="309" alt="Image" src="https://github.com/user-attachments/assets/411ca459-67a0-4c93-a227-d6bb1753d9d0" />
<img width="862" height="309" alt="Image" src="https://github.com/user-attachments/assets/27701a9b-611c-4659-8fa8-6731d7fe4f6f" />

- Ik heb gebruikgemaakt van landmarks zoals header, main en footer:
<img width="873" height="402" alt="Image" src="https://github.com/user-attachments/assets/cc9bd1a5-f06d-44d5-8a3a-898c2eb1a48f" />
<img width="858" height="118" alt="Image" src="https://github.com/user-attachments/assets/a737ff15-c5ae-4f2d-8215-31b890661161" />
<img width="765" height="222" alt="Image" src="https://github.com/user-attachments/assets/c071e207-0835-4acb-a518-9650bc17bcab" />

- Role toegevoegd aan de main:
<img width="665" height="54" alt="Image" src="https://github.com/user-attachments/assets/d6de16ee-ed62-450f-bf47-7ef447ad0363" />

# Nieuwe score
<img width="499" height="212" alt="Image" src="https://github.com/user-attachments/assets/644d340c-b245-4d95-b829-44b8b89bb43f" />
